1차/윤형진_체육복.py

```diff
@@ -0,0 +1,36 @@
+def solution(n, lost, reserve):
+    answer = 0
+    student_list = [0 for i in range(n)]
+    
+    # 여분 보유 학생에 대해 1로 설정, 없는 학생은 -1, 일반 학생은 0
+    # 여분의 체육복을 빌려주면 1을 0으로 바꿈
+    for i in reserve : 
+        student_list[i - 1] = 1 
+        
+    for i in lost :
+        student_list[i - 1] = student_list[i - 1] - 1 #여분을 가져온 학생도 도난당할 수 있다는 점
+
+    for i in range(n) : 
+        if student_list[i] == -1 :
+            if i == 0 : #가장 앞사람인 경우 보다 앞사람이 존재하지 않음
+                if student_list[i + 1] == 1 :
+                    student_list[i + 1] = 0
+                    student_list[i] = 0
+            elif i == (n - 1) : #가장 뒷사람인 경우 보다 뒷사람이 존재하지 않음
+                if student_list[i - 1] == 1 :
+                    student_list[i - 1] = 0
+                    student_list[i] = 0
+            else :
+                if student_list[i - 1] == 1 :
+                    student_list[i - 1] = 0
+                    student_list[i] = 0
+                elif student_list[i + 1] == 1 :
+                    student_list[i + 1] = 0
+                    student_list[i] = 0
```

Comment:
```suggestion
            if i > 0 and student_list[i - 1] == 1 :
                student_list[i - 1] = 0
                student_list[i] = 0
            elif  i < n - 1 and student_list[i + 1] == 1 : 
                student_list[i + 1] = 0
                student_list[i] = 0
```

- 앞 부터 검사를 진행하기 때문에 현재 위치에서 뒤에있는 값을 먼저 검사하는게 최적인 방법이라 생각합니다.
- 이 가정을 통해 코드를 효과적으로 줄일 수 있을 것 같아요.
- 혹시 이 부분에 대해 반례가 있다면 말해주세요 ~

Reply:
참고하겠습니다 감사합니다~~